nipc-openapi/NIPC.yaml

```diff
@@ -0,0 +1,1619 @@
+openapi: 3.0.3
+info:
+  title: Non IP Device Control (NIPC) API
+  description: |-
+    This API specifies RESTful application layer interface for gateways providing operations against non-IP devices. The described interface is extensible. The examples includes leverage Bluetooth Low Energy and Zigbee as they are commonly deployed.
+  termsOfService: http://swagger.io/terms/
+  contact:
+    email: bbrinckm@cisco.com
+  license:
+    name: TBD
+    url: TBD
+  version: 0.4.5
+externalDocs:
+  description: NIPC IETF draft
+  url: https://datatracker.ietf.org/doc/draft-ietf-asdf-nipc/
+servers:
+  - url: "{gw_host}/nipc/draft-04"
+    variables:
+      gw_host:
+        default: localhost
+        description: Gateway Host
+tags:
+  - name: NIPC property APIs
+    description: |-
+      APIs that allow apps to get and update device properties. If the underlying protocol requires connection management, it will be performed as part of the API call.
+  - name: NIPC event APIs
+    description: |-
+      APIs that allow apps to enable or disable event reporting on devices. If the underlying protocol requires connection management, it will be performed as part of the API call.
+  - name: NIPC action APIs
+    description: |-
+      APIs that perform actions on devices.
+  - name: NIPC action APIs with embedded protocol mapping
+    description: |-
+      APIs that perform actions on devices, but do not use registered properties, events or ation. These API's have embedded protocol mappings
+  - name: NIPC registration APIs
+    description: |-
+      APIs that register sdf models or data applications
+  - name:  NIPC API extensions
+    description: |-
+      APIs that simplify application interaction by implementing one or more basic API's into a single API call.
+
+paths:
+### NIPC Property APIs
+  /{id}/property/{property}:
```

Comment:
Are the property names from the SDF definitions? Are there risks for name conflicts (i.e., is there need to use the namespaces)?

Reply:
Yes. There is a risk of name conflicts so we will need to use the absolute uri for the property name. For example, 
`{id}/property/https%3A%2F%2Fexample.com%2Fheartrate%23%2FsdfObject%2Fhealthsensor%2FsdfProperty%2Fheartrate`